nncf/experimental/common/tensor_statistics/collectors.py

```diff
@@ -166,7 +165,38 @@ def _aggregate_impl(self) -> Any:
 
     def reset(self):
         self._collected_samples = 0
-        self._container = []
+        self._reset_sample_container()
+
+    @abstractmethod
+    def _reset_sample_container(self):
+        pass
+
+    def __eq__(self, __o: object) -> bool:
+        return isinstance(__o, self.__class__) and self._num_samples == __o.num_samples
+
+    def __hash__(self) -> int:
+        return hash(self.__class__.__name__)
+
```

Comment:
Please don't do this as this methods are not needed for aggregators but increases responsibility for implementation and testing

Reply:
It seems that I accidentally included the implementation for these methods twice in the class hierarchy. I removed the second implementations, but the first ones remain in the base `Aggregator` class, and it's your own design. Existing code will break if I simply remove thiese methods at all.